core/src/main/clojure/xtdb/buffer_pool.clj

```diff
@@ -36,25 +36,24 @@
   (let [version 2]
     (util/->path (str "v" (util/->lex-hex-string version)))))
 
-(defn- free-memory [^Map memory-store]
-  (locking memory-store
-    (run! util/close (.values memory-store))
-    (.clear memory-store)))
+(defn- free-memory [^Cache memory-store]
+  ;; closing should happen via the remove method
+  (.invalidateAll memory-store)
+  (Thread/sleep 100))
```

Comment:
I spend more time than I am willing to admit trying to get rid of this. Even though we are using a synchronous cache (I think) the invalidation somehow happens async (it seems). Getting at least allocator problems when I don't add this.

Reply:
Might `Cache/.cleanUp` after the invalidation do this?

Reply:
Waitting on the `ForkJoinPool/commonPool` now. This is where things get executed asynchronously.

Reply:
Yep - as discussed, pragmatically let's run with this for now until we can investigate whether there's something we can hook into in Caffeine. Might be an idea to look into their tests to see what they do?